its/src/test/java/com/sonar/it/scanner/msbuild/sonarqube/SolutionKindTest.java

```diff
@@ -49,101 +51,100 @@
+        tuple(SONAR_RULES_PREFIX + "S1186", context.projectKey + ":XamarinApplication/App.xaml.cs"),
+        tuple(SONAR_RULES_PREFIX + "S1186", context.projectKey + ":XamarinApplication/App.xaml.cs"),
+        tuple(SONAR_RULES_PREFIX + "S1186", context.projectKey + ":XamarinApplication/App.xaml.cs"),
+        tuple(SONAR_RULES_PREFIX + "S1134", context.projectKey + ":XamarinApplication/MainPage.xaml.cs"),
+        tuple("external_roslyn:CS0618", context.projectKey + ":XamarinApplication.iOS/Main.cs"));
+
+    assertThat(TestUtils.getMeasureAsInteger(context.projectKey, "lines", context.orchestrator)).isEqualTo(149);
+    assertThat(TestUtils.getMeasureAsInteger(context.projectKey, "ncloc", context.orchestrator)).isEqualTo(93);
+    assertThat(TestUtils.getMeasureAsInteger(context.projectKey, "files", context.orchestrator)).isEqualTo(6);
+    assertThat(TestUtils.getMeasureAsInteger(context.projectKey + ":XamarinApplication.iOS", "lines", context.orchestrator)).isEqualTo(97);
+    assertThat(TestUtils.getMeasureAsInteger(context.projectKey + ":XamarinApplication", "lines", context.orchestrator)).isEqualTo(52);
   }
 
   @Test
   void testRazorCompilationNet9WithoutSourceGenerators() throws IOException {
     assumeTrue(TestUtils.getMsBuildPath(ORCHESTRATOR).toString().contains("2022")); // We can't build without MsBuild17
     String projectName = "RazorWebApplication.net9.withoutSourceGenerators";
-    assertProjectFileContains(projectName, "<UseRazorSourceGenerator>false</UseRazorSourceGenerator>");
-    validateRazorProject(projectName);
+    validateRazorProject(projectName, "<UseRazorSourceGenerator>false</UseRazorSourceGenerator>");
   }
 
   @Test
   void testRazorCompilationNet9WithSourceGenerators() throws IOException {
     assumeTrue(TestUtils.getMsBuildPath(ORCHESTRATOR).toString().contains("2022")); // We can't build without MsBuild17
     String projectName = "RazorWebApplication.net9.withSourceGenerators";
-    assertProjectFileContains(projectName, "<UseRazorSourceGenerator>true</UseRazorSourceGenerator>");
-    validateRazorProject(projectName);
+    validateRazorProject(projectName, "<UseRazorSourceGenerator>true</UseRazorSourceGenerator>");
   }
 
   @Test
   void testCSharpAllFlat() throws IOException {
-    runAnalysis("CSharpAllFlat");
+    var context = AnalysisContext.forServer("CSharpAllFlat");
+    context.build.addArgument("CSharpAllFlat.sln");
+    context.runAnalysis();
 
-    assertThat(getComponent("CSharpAllFlat:Common.cs")).isNotNull();
+    assertThat(getComponent(context.projectKey + ":Common.cs")).isNotNull();
   }
 
   @Test
   void testCSharpSharedFiles() throws IOException {
-    runAnalysis("CSharpSharedFiles");
+    var context = AnalysisContext.forServer("CSharpSharedFiles");
+    context.runAnalysis();
 
-    assertThat(getComponent("CSharpSharedFiles:Common.cs"))
+    assertThat(getComponent(context.projectKey + ":Common.cs"))
```

Comment:
Optional CaYC: It would be nice to remove the useless breaklines in statements like these while we're touching the file